lib/themes/dosomething/paraneue_dosomething/scss/patterns/_button.scss

```diff
@@ -1,14 +1,73 @@
+%animated {
+  animation-fill-mode: both;
+}
+
 // @TODO: Should this become a real modifier?
-.button.-mini {
-  font-size: 13px !important;
-  padding-top: 6px !important;
-  padding-bottom: 4px !important;
-  background: $light-gray;
-  color: $dark-gray;
-  text-transform: none;
-
-  &.is-active {
-    background-color: $purple;
-    color: #fff;
+.button {
+  border: 1px solid $blue;
+  position: relative;
+
+  &.-mini {
+    font-size: 13px !important;
+    padding-top: 6px !important;
+    padding-bottom: 4px !important;
+    background: $light-gray;
+    color: $dark-gray;
+    text-transform: none;
+
+    &.is-active {
+      background-color: $purple;
+      color: #fff;
+    }
+  }
+
+  &.is-loading {
+    border-color: #eee;
+  }
+
+  &.-reactive {
+    &::after {
+      content: '\02713';
+      display: block;
+      font-size: $font-large;
+      left: 50%;
+      line-height: 1;
+      opacity: 0;
+      position: absolute;
+      text-align: center;
+      transform: translate(-50%, -50%);
```

Comment:
it's a trick for centering both vertically and horizontally when you don't know the width & height.